src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java

```diff
@@ -1,60 +1,49 @@
 package com.atlassian.jira.cloud.jenkins.buildinfo.service;
 
-import static java.util.Objects.requireNonNull;
-
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
-
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import com.atlassian.jira.cloud.jenkins.auth.AccessTokenRetriever;
+import com.atlassian.jira.cloud.jenkins.buildinfo.client.BuildsApi;
 import com.atlassian.jira.cloud.jenkins.buildinfo.client.model.BuildApiResponse;
 import com.atlassian.jira.cloud.jenkins.buildinfo.client.model.Builds;
-import com.atlassian.jira.cloud.jenkins.common.client.JiraApi;
-import com.atlassian.jira.cloud.jenkins.common.client.PostUpdateResult;
-import com.atlassian.jira.cloud.jenkins.common.config.JiraSiteConfigRetriever;
-import com.atlassian.jira.cloud.jenkins.common.model.AppCredential;
+import com.atlassian.jira.cloud.jenkins.common.client.ApiUpdateFailedException;
+import com.atlassian.jira.cloud.jenkins.common.config.JiraSiteConfig2Retriever;
 import com.atlassian.jira.cloud.jenkins.common.response.JiraCommonResponse;
 import com.atlassian.jira.cloud.jenkins.common.response.JiraSendInfoResponse;
-import com.atlassian.jira.cloud.jenkins.config.JiraCloudSiteConfig;
+import com.atlassian.jira.cloud.jenkins.config.JiraCloudSiteConfig2;
 import com.atlassian.jira.cloud.jenkins.tenantinfo.CloudIdResolver;
 import com.atlassian.jira.cloud.jenkins.util.Constants;
 import com.atlassian.jira.cloud.jenkins.util.RunWrapperProvider;
 import com.atlassian.jira.cloud.jenkins.util.SecretRetriever;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
 /**
  * Implementation of JiraBuildInfoSender to send build updates to Jira by building the payload,
  * generating the access token, sending the request and parsing the response.
  */
 public abstract class JiraBuildInfoSenderImpl implements JiraBuildInfoSender {
 
-    private static final Logger log = LoggerFactory.getLogger(JiraBuildInfoSenderImpl.class);
-
-    private final JiraSiteConfigRetriever siteConfigRetriever;
+    private final JiraSiteConfig2Retriever siteConfigRetriever;
     private final SecretRetriever secretRetriever;
 
     private final CloudIdResolver cloudIdResolver;
-    private final AccessTokenRetriever accessTokenRetriever;
-    private final JiraApi buildsApi;
+    private final BuildsApi buildsApi;
     protected final RunWrapperProvider runWrapperProvider;
 
     public JiraBuildInfoSenderImpl(
-            final JiraSiteConfigRetriever siteConfigRetriever,
+            final JiraSiteConfig2Retriever siteConfigRetriever,
```

Comment:
We're now loading the configuration from a `JiraSiteConfig2` object instead of a `JiraSiteConfig` object. The new object contains the `webhookUrl`.